dlog/kimchi/src/index.rs

```diff
@@ -246,29 +256,56 @@ where
                 h.insert(Index(GateType::Generic));
                 h
             };
-            let expr = poseidon::constraint();
-            let expr = expr + varbasemul::constraint(super::range::MUL.start);
-            let (alphas_used, complete_add) =
-                complete_add::constraint(super::range::COMPLETE_ADD.start);
-            assert_eq!(alphas_used, super::range::COMPLETE_ADD.len());
-            let expr = expr + complete_add;
-            let expr = expr + endosclmul::constraint(2 + super::range::ENDML.start);
-            let expr = expr + endomul_scalar::constraint(super::range::ENDOMUL_SCALAR.start);
-            let expr = if lookup_used.is_some() {
-                expr + Expr::combine_constraints(
-                    2 + super::range::CHACHA.end,
-                    lookup::constraints(&cs.dummy_lookup_values[0], cs.domain.d1),
-                )
-            } else {
-                expr
-            };
-            let expr = if cs.chacha8.is_some() {
-                expr + chacha::constraint(super::range::CHACHA.start)
-            } else {
-                expr
-            };
 
-            expr.linearize(evaluated_cols).unwrap()
+            // generic gate
+            let mut alphas = powers_of_alpha.register(ConstraintType::Gate(GateType::Generic), 1);
+            alphas.next();
+
+            // permutation
+            let mut alphas = powers_of_alpha.register(ConstraintType::Permutation, 3);
+            alphas.next();
+            alphas.next();
+            alphas.next();
+
+            // poseidon gate
+            let mut alphas = powers_of_alpha.register(ConstraintType::Gate(GateType::Poseidon), 15);
+            let mut expr = poseidon::constraint(&mut alphas);
```

Comment:
Poseidon should be using [0, 15)

Reply:
btw, this is because poseidon is the largest poly?